### PR TITLE
Add kotlin math

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: runtime/src/main/kotlin/konan/internal/RuntimeUtils.kt [hunk 1]
```diff
@@ -61,6 +61,11 @@ fun ThrowUninitializedPropertyAccessException(): Nothing {
     throw UninitializedPropertyAccessException()
 }
 
+@ExportForCppRuntime
+internal fun ThrowNotImplementedError(): Nothing {
```
Comment (PavelPunegov) [inline runtime/src/main/kotlin/konan/internal/RuntimeUtils.kt]: Is it used anywhere?

File: runtime/src/main/cpp/Math.cpp [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+KInt Kotlin_math_absi(KInt x) { return labs(x); }
+KLong Kotlin_math_absl(KLong x) { return llabs(x); }
+
+// endregion
+
+#else // if KONAN_WASM is defined
+
+// kotlin.math isn't supported for WASM.
+namespace {
+    void NotImplemented() { RuntimeAssert(false, "Not implemented for wasm"); }
```
Comment (olonho) [inline runtime/src/main/cpp/Math.cpp]: ThrowNotImplemented()?

File: runtime/src/main/cpp/Math.cpp [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+            }
+            return SQRT2 * result;
+        }
+    }
+}
+#endif
+
+extern "C" {
+
+#ifndef KONAN_WASM
```
Comment (olonho) [inline runtime/src/main/cpp/Math.cpp]: Usually better avoid inverted condition. Better use #if FEATURE, not #if PLATFORM.
So maybe `#if !KONAN_NO_MATH`  here, as on most platforms math.h will be there.

File: runtime/src/main/cpp/Math.cpp [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+ */
+
+#include <float.h>
+#include <math.h>
+#include <stdlib.h>
+
+#include "Exceptions.h"
+#include "Types.h"
+
+#if (__MINGW32__ || __MINGW64__)
```
Comment (olonho) [inline runtime/src/main/cpp/Math.cpp]: Again, smth like `#if KONAN_NEED_ASINH_ACOSH`

**Pull request merged**